### PR TITLE
optimize sec black list

### PR DESCRIPTION
**Pull request by leizhiyuan**
use resource file and switcher to define blacklist .so user can set it by themself

use java resource path instead of the absolute path
use serialize.blacklist.file to define it.

### CODE REVIEW

**Comment by ujjboy** [comment]
CI failed.

Add this  to `pom.xml`
```
<build>
        <resources>
            <resource>
                <directory>src/main/resources</directory>
            </resource>
        </resources>
```

File: src/main/java/com/alipay/hessian/internal/InternalNameBlackListFilter.java [hunk 1]
```diff
@@ -112,4 +54,32 @@ public InternalNameBlackListFilter() {
...
+            classLoader = Thread.currentThread().getContextClassLoader();
+        }
+        final URL resource = classLoader.getResource(relativePath);
+        if (resource != null) {
+            File file = new File(resource.getFile());
+            try {
+                Scanner scanner = new Scanner(file);
+                while (scanner.hasNextLine()) {
+                    String line = scanner.nextLine();
+                    result.add(line);
```
Comment (ujjboy) [inline src/main/java/com/alipay/hessian/internal/InternalNameBlackListFilter.java]: Add to list if not blank.

File: src/main/java/com/alipay/hessian/internal/InternalNameBlackListFilter.java [hunk 2]
```diff
@@ -112,4 +54,32 @@ public InternalNameBlackListFilter() {
...
+                }
+                scanner.close();
```
Comment (ujjboy) [inline src/main/java/com/alipay/hessian/internal/InternalNameBlackListFilter.java]: move to `finally`.

Comment (leizhiyuan) [reply src/main/java/com/alipay/hessian/internal/InternalNameBlackListFilter.java]: ok

**Pull request merged**